### PR TITLE
Only store explicitly set table properties as index settings

### PR DESCRIPTION
**Pull request by mfussenegger**
Relates to https://github.com/crate/crate/issues/11939

And it has the advantage that it is easier to see what settings users configured explicitly on a glance.
In case we change defaults for settings were we think changing the value also for tables created in older versions would be problematic we can in the future then still use old defaults for old versions given that we'll be able to distinguish between explicit/implicit values.

### CODE REVIEW

File: docs/general/ddl/show-create-table.rst [hunk 1]
```diff
@@ -18,52 +18,30 @@ The ``SHOW CREATE TABLE`` statement prints the ``CREATE TABLE`` statement of an
...
+    |    "merge.scheduler.max_thread_count" = 1,    |
+    |    number_of_replicas = '0-1'                 |
+    | )                                             |
+    +-----------------------------------------------+
     SHOW 1 row in set (... sec)
 
-The table settings returned within the ``WITH`` clause of the result are all
-available table settings showing their respective values at the time of the
-execution of the ``SHOW`` statement.
+The ``WITH`` clause always shows ``column_policy`` and ``number_of_replicas``
```
Comment (mfussenegger) [inline docs/general/ddl/show-create-table.rst:43]: `column_policy` is currently derived from `DocTableInfo#columnPolicy`  - in order to distinguish explicit/implicit we'd have to make that nullable which is imho not nice, so I listed this as an exception.

**Review by seut — APPROVED** [review]
👍 But I think we should mention this change in the release notes.

**Comment by mfussenegger** [comment]
> 👍 But I think we should mention this change in the release notes.

Good point. Added an entry

**Comment by mfussenegger** [comment]
retest this please

```
org.opentest4j.AssertionFailedError: 

expected: "GREEN"
 but was: "YELLOW"
	at __randomizedtesting.SeedInfo.seed([1408ACE1FA2450F6:3ADE5F3297DC5EE4]:0)
	at io.crate.integrationtests.LogicalReplicationITestCase.lambda$ensureGreenOnSubscriber$1(LogicalReplicationITestCase.java:267)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:710)
	at io.crate.integrationtests.LogicalReplicationITestCase.ensureGreenOnSubscriber(LogicalReplicationITestCase.java:263)
	at io.crate.integrationtests.LogicalReplicationITestCase.createSubscription(LogicalReplicationITestCase.java:303)
	at io.crate.integrationtests.MetadataTrackerITest.test_subscription_to_multiple_publications_should_not_stop_on_a_single_publication_drop(MetadataTrackerITest.java:267)
```

**Pull request merged**